docs/src/references/c8y-configuration-management.md

```diff
@@ -0,0 +1,162 @@
+# Device Configuration Management using Cumulocity
+
+Thin-edge provides an operation plugin to
+[manage device configurations using Cumulocity](https://cumulocity.com/guides/users-guide/device-management/#to-retrieve-and-apply-a-configuration-snapshot-to-a-device-which-supports-multiple-configuration-types).
+
+* This management is bi-directional:
+  * A device can be taken as reference, all the managed files being uploaded to the cloud
+    and stored there as a configuration snapshot.
+  * A configuration snapshot can be pushed from the cloud to any devices of the same type,
+    i.e. supporting the same kind of configuration files.
+* With this operation plugin, the device owner defines the list of files
+  (usually configuration files, but not necessarily),
+  that will be managed from the cloud tenant.
+* Notably, the plugin configuration itself is managed from the cloud,
+  meaning, the device owner can update from the cloud the list of files to be managed.
+* By default, the managed files are represented on the cloud by their full path on the device.
+  Optionally, the device owner can assign an alias to each target path.  
+* When files are downloaded from the cloud to the device,
+  these files are stored in a temporary directory first.
+  They are atomically moved to their target path, only after a fully successful download.
+  The aim is to avoid breaking the system with half downloaded files.
+* Once all the files of a snapshot have been downloaded,
```

Comment:
"All the files of a snapshot" -> What do you mean? I thought each individual config file is considered as independent units with their individual config snapshots. Is there a concept of multi-file snapshot? Isn't one c8y operation per file and **NOT** a group of files?

Reply:
I agree to you @albinsuresh. The C8Y operation does provide exactly one file per operation:
   https://cumulocity.com/guides/reference/smartrest-two/#download-configuration-file-with-type-524

So I would say we shall remove the term *"all"* in the spec.

Reply:
It was my understanding that a config from the Cumulocity perspective is made of several files. @cstoidner you even added a requirement that the downloads have to be done sequentially.

If there is a single file for each operation, then perfect. This is even simpler.